plugins/tiddlywiki/multiwikiserver/modules/store/sql-tiddler-store.js

```diff
@@ -275,6 +276,10 @@ SqlTiddlerStore.prototype.saveBagTiddlerWithAttachment = function(incomingTiddle
 Returns {tiddler_id:,bag_name:}
 */
 SqlTiddlerStore.prototype.saveRecipeTiddler = function(incomingTiddlerFields,recipe_name) {
+	var tiddlerInfo = this.sqlTiddlerDatabase.getRecipeTiddler(incomingTiddlerFields.title,recipe_name);
+	if (incomingTiddlerFields.type?.includes('image') && tiddlerInfo?.attachment_blob) {
```

Comment:
The check for "image" within the content type would mean that the attachment handling wouldn't work for (say) videos. There's already code further up this file that identifies whether a tiddler qualifies to be handled as an attachment, and so we should generalise that code.